### PR TITLE
Ftr 4499 matriculation link to studies

### LINKED ISSUE #4499: Matriculation link to studies YO-view
Set the link to show when the time is right and the planets are in the correct position.

### PR DESCRIPTION
**Pull request by avukkonen**
Closes #4499

### CODE REVIEW

File: muikku-core-plugins/src/main/resources/fi/otavanopisto/muikku/plugins/transcriptofrecords/TranscriptofRecordsJsPlugin_en.properties [hunk 1]
```diff
@@ -41,7 +41,10 @@ plugin.records.summary.notifications.title = Notifications
 plugin.records.yo.title = Your matricular examinaition
 plugin.records.yo.abiStatus.title = ABI-status
 plugin.records.yo.abiStatus.content.finished = Your ABI-status is does not lack courses. You can now apply for matricular examination.
-plugin.records.yo.button.signUp = Apply for matricular examination
+plugin.records.yo.button.signUp.active = Apply for matricular examination
+plugin.records.yo.button.signUp.active.title = Application ends {0}
```
Comment (jjlankinen) [inline muikku-core-plugins/src/main/resources/fi/otavanopisto/muikku/plugins/transcriptofrecords/TranscriptofRecordsJsPlugin_en.properties]: Should/could be "Application for matriculation examination closes" or "Applying for matriculation examination ends".

File: muikku-core-plugins/src/main/resources/fi/otavanopisto/muikku/plugins/transcriptofrecords/TranscriptofRecordsJsPlugin_en.properties [hunk 2]
```diff
@@ -41,7 +41,10 @@ plugin.records.summary.notifications.title = Notifications
...
+plugin.records.yo.button.signUp.inactive.title = Today is {0}
+plugin.records.yo.button.signUp.inactive = Applying starts at {0}
```
Comment (jjlankinen) [inline muikku-core-plugins/src/main/resources/fi/otavanopisto/muikku/plugins/transcriptofrecords/TranscriptofRecordsJsPlugin_en.properties]: Should/could be "Application for matriculation examination opens" or "Applying for matriculation examination starts".

File: muikku-core-plugins/src/main/resources/fi/otavanopisto/muikku/plugins/transcriptofrecords/TranscriptofRecordsJsPlugin_fi.properties [hunk 3]
```diff
@@ -41,7 +41,10 @@ plugin.records.summary.notifications.title = Ilmoitukset
 plugin.records.yo.title = Ylioppilaskirjoituksesi
 plugin.records.yo.abiStatus.title = ABI-status
 plugin.records.yo.abiStatus.content.finished = ABI-statuksestasi ei puutu kursseja. Voit nyt ilmoittautua YO-kokeeseen.
-plugin.records.yo.button.signUp = Ilmoittaudu Yo-kokeeseen
+plugin.records.yo.button.signUp.active = Ilmoittaudu YO-kokeeseen
+plugin.records.yo.button.signUp.active.title = Ilmoittautuminen p��ttyy {0}
+plugin.records.yo.button.signUp.inactive = Ilmoittautuminen alkaa {0}
+plugin.records.yo.button.signUp.inactive.title = P�iv�m��r� t�n��n: {0}
```
Comment (jjlankinen) [inline muikku-core-plugins/src/main/resources/fi/otavanopisto/muikku/plugins/transcriptofrecords/TranscriptofRecordsJsPlugin_fi.properties]: Lines 46 and 47 should switch so they match the EN locale files.

File: muikku-core-plugins/src/main/resources/fi/otavanopisto/muikku/plugins/transcriptofrecords/TranscriptofRecordsJsPlugin_en.properties [hunk 4]
```diff
@@ -41,7 +41,10 @@ plugin.records.summary.notifications.title = Notifications
 plugin.records.yo.title = Your matricular examinaition
 plugin.records.yo.abiStatus.title = ABI-status
 plugin.records.yo.abiStatus.content.finished = Your ABI-status is does not lack courses. You can now apply for matricular examination.
```
Comment (jjlankinen) [inline muikku-core-plugins/src/main/resources/fi/otavanopisto/muikku/plugins/transcriptofrecords/TranscriptofRecordsJsPlugin_en.properties]: Not part of this PR but should be fixed nevertheless. Suggested change: *You have achieved ABI-status and can now apply for matriculation examination.*

File: muikku-core-plugins/src/main/resources/fi/otavanopisto/muikku/plugins/transcriptofrecords/TranscriptofRecordsJsPlugin_en.properties [hunk 5]
```diff
@@ -41,7 +41,10 @@ plugin.records.summary.notifications.title = Notifications
 plugin.records.yo.title = Your matricular examinaition
```
Comment (jjlankinen) [inline muikku-core-plugins/src/main/resources/fi/otavanopisto/muikku/plugins/transcriptofrecords/TranscriptofRecordsJsPlugin_en.properties]: Not part of this PR but should be changed. Suggested change: Matriculation examination

File: muikku-core-plugins/src/main/resources/fi/otavanopisto/muikku/plugins/transcriptofrecords/TranscriptofRecordsJsPlugin_en.properties [hunk 6]
```diff
@@ -41,7 +41,10 @@ plugin.records.summary.notifications.title = Notifications
...
 plugin.records.yo.abiStatus.title = ABI-status
 plugin.records.yo.abiStatus.content.finished = Your ABI-status is does not lack courses. You can now apply for matricular examination.
-plugin.records.yo.button.signUp = Apply for matricular examination
+plugin.records.yo.button.signUp.active = Apply for matricular examination
+plugin.records.yo.button.signUp.active.title = Application ends {0}
+plugin.records.yo.button.signUp.inactive.title = Today is {0}
+plugin.records.yo.button.signUp.inactive = Applying starts at {0}
 plugin.records.yo.participation.title = Participating in matricular examination
```
Comment (jjlankinen) [inline muikku-core-plugins/src/main/resources/fi/otavanopisto/muikku/plugins/transcriptofrecords/TranscriptofRecordsJsPlugin_en.properties]: Not part of this PR but should be changed. Suggested change: Participating in matriculation examination

File: muikku-core-plugins/src/main/resources/fi/otavanopisto/muikku/plugins/transcriptofrecords/TranscriptofRecordsJsPlugin_fi.properties (hunk 3 above)
Comment (avukkonen) [reply muikku-core-plugins/src/main/resources/fi/otavanopisto/muikku/plugins/transcriptofrecords/TranscriptofRecordsJsPlugin_fi.properties]: rather  EN locale order is wrong

Comment (avukkonen) [reply muikku-core-plugins/src/main/resources/fi/otavanopisto/muikku/plugins/transcriptofrecords/TranscriptofRecordsJsPlugin_fi.properties]: Was
